include/nuttx/sched_note.h

```diff
@@ -700,8 +636,7 @@ void sched_note_filter_tag(FAR struct note_filter_tag_s *oldf,
 }
 #endif
 
-#else /* CONFIG_SCHED_INSTRUMENTATION */
-
+#ifndef CONFIG_SCHED_INSTRUMENTATION
```

Comment:
why replace #else with #ifndef

Reply:
This is to simplify the header file. It uses too many macros, and many of them are unnecessary.